packages/devtools/lib/src/memory/memory_inbounds.dart

```diff
@@ -0,0 +1,347 @@
+    final treeNode02 = InboundsTreeNode('class_0_2', 'field_2');
+    final treeNode03 = InboundsTreeNode('class_0_3', 'field_3');
+    final treeNode04 = InboundsTreeNode('class_0_4', 'field_4');
+    final treeNode05 = InboundsTreeNode('class_0_5', 'field_5');
+
+    final treeNode10 = InboundsTreeNode('class_1', 'field_a');
+    final treeNode11 = InboundsTreeNode('class_1_1', 'field_b');
+    final treeNode12 = InboundsTreeNode('class_1_2', 'field_c');
+    final treeNode13 = InboundsTreeNode('class_1_3', 'field_d');
+    final treeNode14 = InboundsTreeNode('class_4_4', 'field_e');
+    final treeNode15 = InboundsTreeNode('class_1_5', 'field_f');
+
+    final terryStuff = InboundsTreeNode(
+        'TerryStuff allocated TerryExtra', 'extra [object/14752]');
+    final shrineAppState1 = InboundsTreeNode('_ShrineAppState', '_stuff');
+    final shrineAppState2 = InboundsTreeNode('_ShrineAppState', '_stuff2');
+    final statefulElement1 = InboundsTreeNode('StatefulElement', 'state');
+    final hashmapEntry1 = InboundsTreeNode('_HashMapEntry', '_key');
+    final singleChildrenObjectElement1 =
+        InboundsTreeNode('SingleChildrenObjectElement', '_child');
+    final statefulElement2 = InboundsTreeNode('StatefulElement', 'state');
+    final hashmapEntry2 = InboundsTreeNode('_HashMapEntry', '_key');
+    final singleChildrenObjectElement2 =
+        InboundsTreeNode('SingleChildrenObjectElement', '_child');
+
+    data = InboundsTreeNode.root()
+      ..addChild(treeNode00
+        ..addChild(treeNode01)
+        ..addChild(treeNode02
+          ..addChild(treeNode03)
+          ..addChild(treeNode04)
+          ..addChild(treeNode05)))
+      ..addChild(treeNode10
+        ..addChild(treeNode11)
+        ..addChild(treeNode12)
+        ..addChild(treeNode13)
+        ..addChild(treeNode14)
+        ..addChild(treeNode15))
+      ..addChild(terryStuff
+        ..addChild(shrineAppState1
+          ..addChild(statefulElement1
+            ..addChild(hashmapEntry1..addChild(singleChildrenObjectElement1))))
+        ..addChild(shrineAppState2
+          ..addChild(statefulElement2
+            ..addChild(
+                hashmapEntry2..addChild(singleChildrenObjectElement2)))));
+  }
+
+  InboundsTreeNode data;
+}
+
+class InboundsTreeNode extends TreeNode<InboundsTreeNode> {
+  InboundsTreeNode(this._name, this._fieldName, [this.instanceHashCode]);
+  InboundsTreeNode.instance(this._instance, [this.instanceHashCode])
+      : _name = _instance.objectRef,
+        _fieldName = '';
+  InboundsTreeNode.root()
+      : _name = 'Instances',
+        _fieldName = '',
+        instanceHashCode = null;
+  InboundsTreeNode.empty()
+      : _name = null,
+        _fieldName = null,
+        instanceHashCode = null;
+
+  String _name;
+
+  InstanceSummary _instance;
+
+  InstanceSummary get instance => _instance;
+
+  String get name => _name;
+
+  /// isNew signals objectRef (object/###) has changed.
```

Comment:
Consider expanding to follow the Dart doc pattern rather than start with a single detail.

https://dart.dev/guides/language/effective-dart/documentation#do-start-doc-comments-with-a-single-sentence-summary

Reply:
Good idea.

  /// Replaces the node's [instance], [instanceHashCode] and [name].  This is
  /// can happen as a result of the objectRef id changing (as known by the VM).
  /// It's matched to the propery objectRef (instance) by comparing hashCodes.
  ///
  /// [isNew] signals the objectRef (e.g., objects/123) changed to something
  /// different (e.g., objects/245).